webapp/channels/src/utils/url.tsx

```diff
@@ -168,6 +167,24 @@ export function validateChannelUrl(url: string, intl?: IntlShape): Array<React.R
     return errors;
 }
 
+// Returns true when the URL should not trigger any external requests.
+// Currently returns true only for permalinks
+export function doesNotTriggerExternalRequests(url: string, siteURL?: string): boolean {
+    if (siteURL && siteURL !== '') {
+        let standardSiteURL = siteURL;
+        if (standardSiteURL[standardSiteURL.length - 1] === '/') {
+            standardSiteURL = standardSiteURL.substring(0, standardSiteURL.length - 1);
+        }
+        if (!url.startsWith(standardSiteURL)) {
```

Comment:
I wonder if there's some way to simplify this using `new URL(url, siteURL)`, confirming that the URL starts with `siteURL` and then using the `path`. I'm not sure though

Reply:
Probably, but I think just doing string operations is more clear.